send-emails-from-a-service/main.bal

```diff
@@ -0,0 +1,173 @@
+
+    resource function post categories/[string category]/reserve(ReservationRequest reservationRequest)
+            returns http:InternalServerError|http:NotFound|http:Ok {
+
+        email:SmtpClient|email:Error smtpClient = new (host, senderAddress, appPassword);
+        
+        if smtpClient is email:Error {
+            return <http:InternalServerError> {body: smtpClient.message()};
+        } 
+
+        ReservationRequest {
+            patient: {cardNo, ...patient},
+            doctor,
+            hospital,
+            hospital_id,
+            appointment_date
+        } = reservationRequest;
+
+        Appointment|http:ClientError appointment =
+                hospitalServicesEP->/[hospital_id]/categories/[category]/reserve.post({
+            patient,
+            doctor,
+            hospital,
+            appointment_date
+        });
+
+        if appointment !is Appointment {
+            log:printError("Appointment reservation failed", appointment);
+            if appointment is http:ClientRequestError {
+                return <http:NotFound> {body: string `unknown hospital, doctor, or category`};
+            }
+            return <http:InternalServerError> {body: appointment.message()};
+        }
+
+        int appointmentNumber = appointment.appointmentNumber;
+
+        PaymentSettlement paymentSettlement = {
+            appointmentNumber: appointmentNumber,
+            doctor: appointment.doctor,
+            patient: appointment.patient,
+            fee: appointment.doctor.fee,
+            confirmed: appointment.confirmed,
+            card_number: cardNo
+        };
+
+        Payment|http:ClientError payment =
+                hospitalServicesEP->/healthcare/payments.post(paymentSettlement);
+
+        if payment !is Payment {
+            log:printError("Payment settlement failed", payment);
+            if payment is http:ClientRequestError {
+                return <http:NotFound> {body: string `unknown hospital, patient, or category`};
+            }
+            return <http:InternalServerError> {body: payment.message()};
+        }
+
+        ReservationResponse reservationResponse = {
+            appointmentNo: appointmentNumber,
+            doctorName: reservationRequest.doctor,
+            patient: reservationRequest.patient.name,
+            actualFee: payment.actualFee,
+            discount: payment.discount,
+            discounted: payment.discounted,
+            paymentID: payment.paymentID,
+            status: payment.status
+        };
+
+        string[] messageContent = from var [key, value] in reservationResponse.entries()
+            let string str = key + ": " + value.toString()
+            select str;
+
+        string emailBody = string:'join("\n", ...messageContent);
+
+        email:Message email = {
+            to: reservationRequest.patient.email,
+            subject: "Payment Status",
+            body: emailBody
+        };
```

Comment:
```suggestion
        string[] messageContent = from var [key, value] in reservationResponse.entries()
            select key + ": " + value.toString();

        email:Message email = {
            to: reservationRequest.patient.email,
            subject: "Payment Status",
            body: string:'join("\n", ...messageContent)
        };
```

But, I would rather we use a string template and come up with a proper body. Like something you would get in a real-world email.